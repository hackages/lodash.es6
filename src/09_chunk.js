/**
 * Creates an array of elements split into groups the length of size. If array can't be split evenly, the final chunk
 * will be the remaining elements.
 */
export function chunk(array, size = 1) {}
