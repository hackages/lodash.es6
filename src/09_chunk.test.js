import {chunk} from './09_chunk'

describe('chunk', () => {
  it('should chunk', () => {
    expect(
      chunk(['a', 'b', 12, true, false, undefined, null, 'aa'], 1)
    ).toMatchSnapshot()

    expect(
      chunk(['a', 'b', 12, true, false, undefined, null, 'aa'], 2)
    ).toMatchSnapshot()

    expect(
      chunk(['a', 'b', 12, true, false, undefined, null, 'aa'], 3)
    ).toMatchSnapshot()

    expect(chunk([], 2)).toMatchSnapshot()

    expect(chunk([42], 2)).toMatchSnapshot()

    expect(chunk(undefined, 4)).toMatchSnapshot()

    expect(chunk(32, 2)).toMatchSnapshot()

    expect(chunk({a: 12, b: 32, c: 45}, 2)).toMatchSnapshot()
  })
})
