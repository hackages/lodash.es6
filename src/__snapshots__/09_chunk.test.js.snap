// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chunk should chunk 1`] = `
Array [
  Array [
    "a",
  ],
  Array [
    "b",
  ],
  Array [
    12,
  ],
  Array [
    true,
  ],
  Array [
    false,
  ],
  Array [
    undefined,
  ],
  Array [
    null,
  ],
  Array [
    "aa",
  ],
]
`;

exports[`chunk should chunk 2`] = `
Array [
  Array [
    "a",
    "b",
  ],
  Array [
    12,
    true,
  ],
  Array [
    false,
    undefined,
  ],
  Array [
    null,
    "aa",
  ],
]
`;

exports[`chunk should chunk 3`] = `
Array [
  Array [
    "a",
    "b",
    12,
  ],
  Array [
    true,
    false,
    undefined,
  ],
  Array [
    null,
    "aa",
  ],
]
`;

exports[`chunk should chunk 4`] = `Array []`;

exports[`chunk should chunk 5`] = `
Array [
  Array [
    42,
  ],
]
`;

exports[`chunk should chunk 6`] = `Array []`;

exports[`chunk should chunk 7`] = `Array []`;

exports[`chunk should chunk 8`] = `Array []`;
